src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/TypePreinit.cs

```diff
@@ -1864,12 +1944,49 @@ private static BaseValueTypeValue NewUninitializedLocationValue(TypeDesc locatio
             }
         }
 
-        private bool TryHandleIntrinsicCall(MethodDesc method, Value[] parameters, out Value retVal)
+        private bool TryHandleIntrinsicCall(TypeSystemContext context, MethodDesc method, Value[] parameters, out Value retVal)
         {
             retVal = default;
 
             switch (method.Name)
             {
+                case "Memmove":
+                    if (method.OwningType is MetadataType spanHelpersType
+                        && spanHelpersType.Name == "SpanHelpers" && spanHelpersType.Namespace == "System"
+                        && spanHelpersType.Module == spanHelpersType.Context.SystemModule
+                        && parameters[0] is ByRefValue dest
+                        && parameters[1] is ByRefValue src
+                        && parameters[2] is ValueTypeValue len)
+                    {
+                        int length = context.Target.PointerSize == 8 ? checked ((int)len.AsInt64()) : len.AsInt32();
```

Comment:
if it overflows we then just give up on preinitializing, right?

Reply:
Good spot actually, i need to catch that, the big outer `catch (TypeSystemException ex)` would not catch that exception